File: codingTest/programmers/구현/K번째수.go
// K번쨰수.go
// https://programmers.co.kr/learn/courses/30/lessons/42748?language=go

package main

import (
	"fmt"
	"sort"
)

func solution(array []int, commands [][]int) []int {
	answer := make([]int, 0)
	for _, command := range commands {
		slice := make([]int, command[1]-command[0]+1)
		copy(slice, array[command[0]-1:command[1]])
		sort.Ints(slice)
		answer = append(answer, slice[command[2]-1])
	}
	return answer
}

func main() {
	array := []int{1, 5, 2, 6, 3, 7, 4}
	commands := [][]int{{2, 5, 3}, {4, 4, 1}, {1, 7, 3}}
	fmt.Println(solution(array, commands))
}

File: codingTest/programmers/정렬/K번째수.py
def solution(array, commands):
    answer = []
    for i in range(len(commands)) :
        tmp = array[commands[i][0]-1:commands[i][1]]
        tmp.sort()
        answer.append(tmp[commands[i][2]-1])
    return answer
print(solution([1, 5, 2, 6, 3, 7, 4],[[2, 5, 3], [4, 4, 1], [1, 7, 3]]))

def solution(array, commands):
    answer = []
    
    for cmd in commands:
        answer.append(sorted(array[cmd[0]-1:cmd[1]])[cmd[2] - 1])
    return answer